File: javascript/src/models/ingestMessageAttemptExhaustedEvent.ts
// this file is @generated
import {
  type IngestMessageAttemptExhaustedEventData,
  IngestMessageAttemptExhaustedEventDataSerializer,
} from "./ingestMessageAttemptExhaustedEventData";

/** Sent when a message delivery has failed (all of the retry attempts have been exhausted). */
export interface IngestMessageAttemptExhaustedEvent {
  data: IngestMessageAttemptExhaustedEventData;
  type: string;
}

export const IngestMessageAttemptExhaustedEventSerializer = {
  _fromJsonObject(object: any): IngestMessageAttemptExhaustedEvent {
    return {
      data: IngestMessageAttemptExhaustedEventDataSerializer._fromJsonObject(
        object["data"]
      ),
      type: object["type"],
    };
  },

  _toJsonObject(self: IngestMessageAttemptExhaustedEvent): any {
    return {
      data: IngestMessageAttemptExhaustedEventDataSerializer._toJsonObject(self.data),
      type: self.type,
    };
  },
};

File: rust/src/models/ingest_message_attempt_exhausted_event.rs
// this file is @generated
use serde::{
    Deserialize,
    Serialize,
};

use super::ingest_message_attempt_exhausted_event_data::IngestMessageAttemptExhaustedEventData;

/// Sent when a message delivery has failed (all of the retry attempts have been
/// exhausted).
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct IngestMessageAttemptExhaustedEvent {
    pub data: IngestMessageAttemptExhaustedEventData,

    pub r#type: String,
}

impl IngestMessageAttemptExhaustedEvent {
    pub fn new(
        data: IngestMessageAttemptExhaustedEventData,
        r#type: String,
    ) -> Self {
        Self {
            data,
            r#type,
        }
    }
}
